core/utils/modelutils.py

```diff
@@ -0,0 +1,149 @@
+    additional_parameters: Any new parameters to be added at the same level of the parameter to be replaced.
+    """
+
+    def change_value(dic, key, value, additional_parameters={}):
+        for k, v in dic.items():
+            if k == key and type(dic[k]) == type(value):
+                dic[k] = value
+                if list(additional_parameters.keys()) !=[]:
+                    for k, v in additional_parameters.items():
+                        dic[k] = v
+            elif isinstance(v, ConfigDict):
+                change_value(v, key, value,additional_parameters)
+
+    for k in config.keys():
+        if isinstance(config[k], ConfigDict):
+            change_value(config[k], config_key, new_value, create_additional_parameters)
+        elif k == config_key:
+            config[k] = new_value
+    return config
+
+
+def get_metainfo_coco(data_path):
+    """
+    OpenMM Libraries requires a modified metainfo to support COCO dataloading for custom datasets.
+    """
+    from pycocotools.coco import COCO
+
+    ann_file = os.path.join(data_path, "annotations/instances_val2017.json")
+    coco = COCO(ann_file)
+    categories = coco.loadCats(coco.getCatIds())
+    category_names = [category["name"] for category in categories]
+    metainfo = dict(classes=category_names)
+    return metainfo
+
+
+def modify_head_classification(model, model_name, num_classes):
+    """
+    Modifies the head of classification models loaded from timm and huggingface to support the number of classes of the loaded custom datasets
+    """
+    import torch
+    import torch.nn as nn
+
+    layer_names = [name for name, _ in model.named_children()]
+    if "head" in layer_names:
+        nc = [i for i in model.head.named_children()]
+        if nc == []:
+            setattr(model, "head", nn.Linear(model.head.in_features, num_classes))
+        else:
+            if "fc" in model.head.named_children():
+                setattr(
+                    model.head, "fc", nn.Linear(model.head.fc.in_features, num_classes)
+                )
+    elif "fc" in layer_names:
+        setattr(model, "fc", nn.Linear(model.fc.in_features, num_classes))
+    elif "classifier" in layer_names:
+        setattr(
+            model, "classifier", nn.Linear(model.classifier.in_features, num_classes)
+        )
+    elif "vanillanet" in model_name:
+        model.switch_to_deploy()
+        model.cls[2] = nn.Conv2d(
+            model.cls[2].in_channels,
+            num_classes,
+            kernel_size=model.cls[2].kernel_size,
+            stride=model.cls[2].stride,
+        )
+    else:
+        raise ValueError(
+            f"Not able to find the last fc layer from the layer list {layer_names}"
+        )
+    return model
+
+
+def find_state_or_model(loaded):
+    """
+    Identifies if loaded model is a state dict or a nn.module
+    """
+    if isinstance(loaded, OrderedDict) or isinstance(loaded, dict):
+        return "STATE_DICT"
+    else:
+        return "MODEL"
```

Comment:
do we need this method to infer this? we can directly perform an isinstance check and derive if a loaded weight is model or state_dict.

if we are keeping this logic, we should also confirm the type for the model
```suggestion
def get_state_dict_or_model(loaded):
    """
    Identifies if loaded file is a state dict or a nn.module
    """
    if isinstance(loaded, OrderedDict, dict):
        return "STATE_DICT"
    elif isinstance(loaded, nn.Module): ## add imports
        return "MODEL"
```

Reply:
made this change